File: java/sprykerFileCreator/ui/validator/FormValidator.java
package sprykerFileCreator.ui.validator;

import pav.sprykerFileCreator.ui.ClassSelectionItem;
import pav.sprykerFileCreator.ui.SprykerFileCreatorForm;

import javax.xml.bind.ValidationException;

public class FormValidator implements FormValidatorInterface {

    public void validate(SprykerFileCreatorForm form) throws ValidationException {
        this.validateFileSelectionItems(form);
        this.validateBundleName(form);
    }

    private void validateFileSelectionItems(SprykerFileCreatorForm form) throws ValidationException {
        for (ClassSelectionItem fileSelectionItem : form.getClassSelectionItems()) {
            if (fileSelectionItem.isSelected()) {
                return;
            }
        }

        throw new ValidationException("blub");
    }

    private void validateBundleName(SprykerFileCreatorForm form) throws ValidationException {
        if (form.getBundleName().isEmpty()) {
            throw new ValidationException("blub");
        }
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/ui/validator/FormValidator.kt
package com.github.tobi812.sprykerplugin.ui.validator

import sprykerFileCreator.ui.SprykerFileCreatorForm
import javax.xml.bind.ValidationException

class FormValidator : FormValidatorInterface {

    @Throws(ValidationException::class)
    override fun validate(form: SprykerFileCreatorForm) {
        this.validateFileSelectionItems(form)
        this.validateBundleName(form)
    }

    @Throws(ValidationException::class)
    private fun validateFileSelectionItems(form: SprykerFileCreatorForm) {
        for (fileSelectionItem in form.getClassSelectionItems()) {
            if (fileSelectionItem.isSelected()) {
                return
            }
        }

        throw ValidationException("blub")
    }

    @Throws(ValidationException::class)
    private fun validateBundleName(form: SprykerFileCreatorForm) {
        if (form.getBundleName().isEmpty()) {
            throw ValidationException("blub")
        }
    }
}